test/tests/val-2f/val-2f.i

```diff
@@ -0,0 +1,268 @@
+  []
+  [flux_x_Be]
+    type = DiffusionFluxAux
+    diffusivity = diffusivity_Be
+    variable = flux_x
+    diffusion_variable = deuterium_concentration_Be
+    component = x
+    block = 1
+  []
+[]
+
+[BCs]
+  [left_flux]
+    type = EquilibriumBC
+    Ko = ${solubility_constant_BeO}
+    activation_energy = '${fparse solubility_energy_BeO * R}'
+    boundary = left
+    enclosure_var = enclosure_pressure
+    temperature = temperature
+    variable = deuterium_concentration_Be
+    p = ${solubility_order}
+  []
+  [right_flux]
+    type = ADNeumannBC
+    boundary = right
+    variable = deuterium_concentration_Be
+    value = 0
+  []
+[]
+
+[Functions]
+  [temperature_bc_func]
+    type = ParsedFunction
+    expression = 'if(t<${charge_time}, ${temperature_initial}, if(t<${fparse charge_time + cooldown_duration}, ${temperature_initial}-((1-exp(-(t-${charge_time})/${cooldown_time_constant}))*${fparse temperature_initial - temperature_cooldown_min}), ${temperature_desorption_min}+${desorption_heating_rate}*(t-${fparse charge_time + cooldown_duration})))'
+  []
+  [diffusivity_Be_func]
+    type = ParsedFunction
+    symbol_names = 'T'
+    symbol_values = 'temperature_bc_func'
+    expression = '${diffusion_Be_preexponential}*exp(-${diffusion_Be_energy}/T)'
+  []
+  [enclosure_pressure_func]
+    type = ParsedFunction
+    expression = 'if(t<${charge_time}, ${pressure_enclosure_init}, if(t<${fparse charge_time + cooldown_duration}, ${pressure_enclosure_cooldown}, ${pressure_enclosure_desorption}))'
+  []
+  [solubility_Be_func]
+    type = ParsedFunction
+    symbol_names = 'T'
+    symbol_values = 'temperature_bc_func'
+    expression = '${solubility_constant_Be} * exp(-${solubility_energy_Be}/T)'
+  []
+  [max_time_step_size_func]
+    type = ParsedFunction
+    expression = 'if(t < ${fparse charge_time-dt_max_large}, ${dt_max_large}, ${dt_max_small})'
+  []
+[]
+
+[Materials]
+  [diffusion_solubility]
+    type = ADGenericFunctionMaterial
+    prop_names = 'diffusivity_Be solubility_Be'
+    prop_values = 'diffusivity_Be_func solubility_Be_func'
+    outputs = all
+  []
+  [converter_to_nonAD]
+    type = MaterialADConverter
+    ad_props_in = 'diffusivity_Be'
+    reg_props_out = 'diffusivity_Be_nonAD'
+  []
+[]
+
+[Postprocessors]
+  [avg_flux_left]
+    type = SideDiffusiveFluxAverage
+    variable = deuterium_concentration_Be
+    boundary = left
+    diffusivity = diffusivity_Be_nonAD
+  []
+  [avg_flux_total] # total flux coming out of the sample in atoms/microns^2/s
+    type = ScalePostprocessor
+    value = avg_flux_left
+    scaling_factor = '${fparse 2 * concentration_scaling}'
+    # Factor of 2 because symmetry is assumed and only one-half of the specimen is modeled.
+    # Thus, the total flux coming out of the specimen (per unit area)
+    # is twice the flux calculated at the left side of the domain.
+    # The 'concentration_scaling' parameter is used to get a consistent concentration unit
+  []
```

Comment:
Is this really relevant in the new validation case? You should check if they capture the flux on one end or on both sides.